File: src/test/java/net/tassia/ini/IniTest.java
package net.tassia.ini;

import net.tassia.io.Streams;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;

import static org.junit.jupiter.api.Assertions.*;

public class IniTest {

	private String getResource(String path) {
		try (InputStream in = IniTest.class.getResourceAsStream(path)) {
			if (in == null) {
				throw new AssertionError("Resource " + path + " not found.");
			}
			byte[] data = Streams.readAllBytes(in);
			return new String(data, StandardCharsets.UTF_8);
		} catch (IOException ex) {
			throw new AssertionError(ex);
		}
	}

	@Test
	public void test1() {
		String str = getResource("/net/tassia/ini/test1.ini");
		IniParser parser = new IniParser(str);
		IniDocument doc = parser.readDocument();

		assertEquals(2, doc.getRoot().size());
		assertEquals(1, doc.getSection("Section1").size());
		assertEquals(1, doc.getSection("Section 2").size());

		assertEquals("Hello World!", doc.getProperty("ROOT_PROPERTY"));
		assertEquals("", doc.getProperty("ROOT_EMPTY_PROPERTY"));

		assertEquals("1", doc.getProperty("Section1.SOME_PROP"));
		assertEquals("2", doc.getProperty("Section 2.SOME_PROP"));
	}

}

File: commons-ini/src/test/java/net/tassia/ini/IniTest.kt
package net.tassia.ini

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import java.io.InputStream

@DisplayName("INI")
class IniTest {

	private fun getResource(path: String): String {
		return IniTest::class.java.getResourceAsStream(path)?.use { it.readAllBytes() }
			?.let { String(it, Charsets.UTF_8) } ?: throw AssertionError("Test resource does not exist: $path")
	}

	@Test
	@DisplayName("Test #1")
	fun test1() {

		val str = getResource("/net/tassia/ini/test1.ini")
		val parser = IniParser(str)
		val doc = parser.readDocument()

		Assertions.assertEquals(2, doc.root.size)

		Assertions.assertEquals(1, doc.getSection("Section1").size)
		Assertions.assertEquals(1, doc.getSection("Section 2").size)

		Assertions.assertEquals("Hello World!", doc.getProperty("ROOT_PROPERTY"))
		Assertions.assertEquals("", doc.getProperty("ROOT_EMPTY_PROPERTY"))

		Assertions.assertEquals("1", doc.getProperty("Section1.SOME_PROP"))
		Assertions.assertEquals("2", doc.getProperty("Section 2.SOME_PROP"))

	}

}
